<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindowClass</class>
 <widget class="QMainWindow" name="MainWindowClass">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>874</width>
    <height>548</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Sphere Marcher</string>
  </property>
  <widget class="QWidget" name="centralWidget">
   <layout class="QGridLayout" name="gridLayout">
    <item row="0" column="0">
     <widget class="QTabWidget" name="tabWidget">
      <property name="tabPosition">
       <enum>QTabWidget::North</enum>
      </property>
      <property name="tabShape">
       <enum>QTabWidget::Triangular</enum>
      </property>
      <property name="currentIndex">
       <number>0</number>
      </property>
      <widget class="QWidget" name="tab">
       <attribute name="title">
        <string>Scene Description Code</string>
       </attribute>
       <layout class="QHBoxLayout" name="horizontalLayout_2">
        <item>
         <widget class="QSplitter" name="splitter_2">
          <property name="orientation">
           <enum>Qt::Horizontal</enum>
          </property>
          <widget class="QTextEdit" name="teCode">
           <property name="styleSheet">
            <string notr="true">background-color: rgb(63, 63, 63);
color: rgb(220, 220, 204);
selection-background-color: rgb(65,54,60);
selection-color: rgb(236,188,188);</string>
           </property>
           <property name="frameShape">
            <enum>QFrame::NoFrame</enum>
           </property>
           <property name="frameShadow">
            <enum>QFrame::Plain</enum>
           </property>
           <property name="documentTitle">
            <string/>
           </property>
          </widget>
          <widget class="QGroupBox" name="groupBox">
           <property name="title">
            <string/>
           </property>
           <layout class="QVBoxLayout" name="verticalLayout_5">
            <item>
             <widget class="QToolBox" name="tbPanel">
              <property name="styleSheet">
               <string notr="true">#toolBox::tab:selected { font: bold; color: #4FA600;height: 26px;} /* selected tab */
#toolBox::tab:!selected { font: bold; color: rgb(85,85,84);height: 26px;} /* non-selected tab */
#toolBox{
border: 0px transparent;
}

#toolBox::tab:QToolButton{
border: 1px transparent;
border-top-left-radius: 6px;
border-top-right-radius: 6px;
background-color: rgb(234, 234, 234); /* Button background color no need to write long code */
}</string>
              </property>
              <property name="currentIndex">
               <number>0</number>
              </property>
              <widget class="QWidget" name="page_4">
               <property name="geometry">
                <rect>
                 <x>0</x>
                 <y>0</y>
                 <width>360</width>
                 <height>377</height>
                </rect>
               </property>
               <attribute name="label">
                <string>Render</string>
               </attribute>
               <layout class="QVBoxLayout" name="verticalLayout_2">
                <item>
                 <widget class="QLabel" name="lPreview">
                  <property name="minimumSize">
                   <size>
                    <width>180</width>
                    <height>180</height>
                   </size>
                  </property>
                  <property name="maximumSize">
                   <size>
                    <width>15000</width>
                    <height>180</height>
                   </size>
                  </property>
                  <property name="styleSheet">
                   <string notr="true"/>
                  </property>
                  <property name="frameShape">
                   <enum>QFrame::StyledPanel</enum>
                  </property>
                  <property name="frameShadow">
                   <enum>QFrame::Plain</enum>
                  </property>
                  <property name="text">
                   <string/>
                  </property>
                  <property name="alignment">
                   <set>Qt::AlignCenter</set>
                  </property>
                 </widget>
                </item>
                <item>
                 <widget class="QCommandLinkButton" name="renderButton">
                  <property name="text">
                   <string>Render Scene</string>
                  </property>
                 </widget>
                </item>
                <item>
                 <spacer name="verticalSpacer_2">
                  <property name="orientation">
                   <enum>Qt::Vertical</enum>
                  </property>
                  <property name="sizeHint" stdset="0">
                   <size>
                    <width>20</width>
                    <height>40</height>
                   </size>
                  </property>
                 </spacer>
                </item>
                <item>
                 <widget class="QLabel" name="lInfo">
                  <property name="text">
                   <string/>
                  </property>
                 </widget>
                </item>
               </layout>
              </widget>
              <widget class="QWidget" name="page_2">
               <property name="geometry">
                <rect>
                 <x>0</x>
                 <y>0</y>
                 <width>158</width>
                 <height>201</height>
                </rect>
               </property>
               <attribute name="label">
                <string>Raymarcher Settings</string>
               </attribute>
               <layout class="QVBoxLayout" name="verticalLayout_3">
                <item>
                 <layout class="QGridLayout" name="gridLayout_2">
                  <item row="0" column="0">
                   <widget class="QLabel" name="label_3">
                    <property name="text">
                     <string>Render Size:</string>
                    </property>
                   </widget>
                  </item>
                  <item row="3" column="0">
                   <widget class="QLabel" name="label">
                    <property name="sizePolicy">
                     <sizepolicy hsizetype="Minimum" vsizetype="Minimum">
                      <horstretch>0</horstretch>
                      <verstretch>0</verstretch>
                     </sizepolicy>
                    </property>
                    <property name="text">
                     <string>Height:</string>
                    </property>
                    <property name="scaledContents">
                     <bool>true</bool>
                    </property>
                   </widget>
                  </item>
                  <item row="1" column="4">
                   <widget class="QSpinBox" name="widthSpin">
                    <property name="minimum">
                     <number>10</number>
                    </property>
                    <property name="maximum">
                     <number>1600</number>
                    </property>
                    <property name="value">
                     <number>600</number>
                    </property>
                   </widget>
                  </item>
                  <item row="1" column="0">
                   <widget class="QLabel" name="label_2">
                    <property name="sizePolicy">
                     <sizepolicy hsizetype="Minimum" vsizetype="Minimum">
                      <horstretch>0</horstretch>
                      <verstretch>0</verstretch>
                     </sizepolicy>
                    </property>
                    <property name="text">
                     <string>Width:</string>
                    </property>
                    <property name="scaledContents">
                     <bool>true</bool>
                    </property>
                   </widget>
                  </item>
                  <item row="3" column="4">
                   <widget class="QSpinBox" name="heightSpin">
                    <property name="minimum">
                     <number>10</number>
                    </property>
                    <property name="maximum">
                     <number>1200</number>
                    </property>
                    <property name="value">
                     <number>320</number>
                    </property>
                   </widget>
                  </item>
                  <item row="8" column="0">
                   <spacer name="verticalSpacer_3">
                    <property name="orientation">
                     <enum>Qt::Vertical</enum>
                    </property>
                    <property name="sizeHint" stdset="0">
                     <size>
                      <width>20</width>
                      <height>40</height>
                     </size>
                    </property>
                   </spacer>
                  </item>
                  <item row="4" column="0">
                   <widget class="Line" name="line">
                    <property name="orientation">
                     <enum>Qt::Vertical</enum>
                    </property>
                   </widget>
                  </item>
                  <item row="5" column="0">
                   <widget class="QLabel" name="label_4">
                    <property name="text">
                     <string>Background color:</string>
                    </property>
                   </widget>
                  </item>
                  <item row="5" column="4">
                   <widget class="QPushButton" name="bkgColorButton">
                    <property name="toolTip">
                     <string>Background color, click to choose</string>
                    </property>
                    <property name="styleSheet">
                     <string notr="true">background-color: rgb(0, 0, 0);</string>
                    </property>
                    <property name="text">
                     <string/>
                    </property>
                    <property name="autoDefault">
                     <bool>false</bool>
                    </property>
                    <property name="flat">
                     <bool>false</bool>
                    </property>
                   </widget>
                  </item>
                  <item row="6" column="0">
                   <widget class="QLabel" name="label_5">
                    <property name="text">
                     <string>Max steps:</string>
                    </property>
                   </widget>
                  </item>
                  <item row="7" column="0">
                   <widget class="QDial" name="maxSteps">
                    <property name="minimum">
                     <number>1</number>
                    </property>
                    <property name="maximum">
                     <number>200</number>
                    </property>
                    <property name="pageStep">
                     <number>10</number>
                    </property>
                    <property name="value">
                     <number>64</number>
                    </property>
                    <property name="orientation">
                     <enum>Qt::Horizontal</enum>
                    </property>
                   </widget>
                  </item>
                  <item row="7" column="4">
                   <widget class="QLCDNumber" name="lcdNumber">
                    <property name="frameShape">
                     <enum>QFrame::NoFrame</enum>
                    </property>
                    <property name="smallDecimalPoint">
                     <bool>false</bool>
                    </property>
                    <property name="numDigits">
                     <number>3</number>
                    </property>
                    <property name="intValue" stdset="0">
                     <number>64</number>
                    </property>
                   </widget>
                  </item>
                 </layout>
                </item>
               </layout>
              </widget>
             </widget>
            </item>
           </layout>
          </widget>
         </widget>
        </item>
       </layout>
      </widget>
      <widget class="QWidget" name="tab_2">
       <attribute name="title">
        <string>Rendered</string>
       </attribute>
       <layout class="QHBoxLayout" name="horizontalLayout">
        <item>
         <widget class="QScrollArea" name="scrollArea">
          <property name="frameShape">
           <enum>QFrame::Box</enum>
          </property>
          <property name="frameShadow">
           <enum>QFrame::Plain</enum>
          </property>
          <property name="verticalScrollBarPolicy">
           <enum>Qt::ScrollBarAsNeeded</enum>
          </property>
          <property name="widgetResizable">
           <bool>true</bool>
          </property>
          <widget class="QWidget" name="scrollAreaWidgetContents">
           <property name="geometry">
            <rect>
             <x>0</x>
             <y>0</y>
             <width>98</width>
             <height>33</height>
            </rect>
           </property>
           <layout class="QVBoxLayout" name="verticalLayout">
            <item>
             <widget class="QLabel" name="lPicture">
              <property name="sizePolicy">
               <sizepolicy hsizetype="Preferred" vsizetype="Preferred">
                <horstretch>0</horstretch>
                <verstretch>0</verstretch>
               </sizepolicy>
              </property>
              <property name="autoFillBackground">
               <bool>true</bool>
              </property>
              <property name="frameShape">
               <enum>QFrame::StyledPanel</enum>
              </property>
              <property name="frameShadow">
               <enum>QFrame::Plain</enum>
              </property>
              <property name="text">
               <string/>
              </property>
              <property name="textFormat">
               <enum>Qt::PlainText</enum>
              </property>
              <property name="scaledContents">
               <bool>false</bool>
              </property>
              <property name="indent">
               <number>0</number>
              </property>
             </widget>
            </item>
           </layout>
          </widget>
         </widget>
        </item>
       </layout>
      </widget>
     </widget>
    </item>
   </layout>
  </widget>
  <widget class="QMenuBar" name="menuBar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>874</width>
     <height>21</height>
    </rect>
   </property>
   <widget class="QMenu" name="menuFile">
    <property name="title">
     <string>File</string>
    </property>
    <addaction name="actionOpen"/>
    <addaction name="actionSave_code"/>
    <addaction name="separator"/>
    <addaction name="actionExit"/>
   </widget>
   <widget class="QMenu" name="menuImage">
    <property name="title">
     <string>Render</string>
    </property>
    <addaction name="actionSave"/>
   </widget>
   <addaction name="menuFile"/>
   <addaction name="menuImage"/>
  </widget>
  <widget class="QStatusBar" name="statusBar"/>
  <action name="actionOpen">
   <property name="text">
    <string>Open</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+O</string>
   </property>
  </action>
  <action name="actionSave_code">
   <property name="text">
    <string>Save</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+S</string>
   </property>
  </action>
  <action name="actionExit">
   <property name="text">
    <string>Exit</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+Q</string>
   </property>
  </action>
  <action name="actionSave">
   <property name="text">
    <string>Save</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+Shift+S</string>
   </property>
  </action>
 </widget>
 <layoutdefault spacing="6" margin="11"/>
 <resources>
  <include location="mainwindow.qrc"/>
 </resources>
 <connections>
  <connection>
   <sender>actionOpen</sender>
   <signal>triggered()</signal>
   <receiver>MainWindowClass</receiver>
   <slot>openCode()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>370</x>
     <y>257</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionSave</sender>
   <signal>triggered()</signal>
   <receiver>MainWindowClass</receiver>
   <slot>saveImage()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>370</x>
     <y>257</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionExit</sender>
   <signal>triggered()</signal>
   <receiver>MainWindowClass</receiver>
   <slot>close()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>370</x>
     <y>257</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionSave_code</sender>
   <signal>triggered()</signal>
   <receiver>MainWindowClass</receiver>
   <slot>saveCode()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>474</x>
     <y>296</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>renderButton</sender>
   <signal>released()</signal>
   <receiver>MainWindowClass</receiver>
   <slot>compileAndRun()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>642</x>
     <y>131</y>
    </hint>
    <hint type="destinationlabel">
     <x>563</x>
     <y>0</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>bkgColorButton</sender>
   <signal>released()</signal>
   <receiver>MainWindowClass</receiver>
   <slot>setBkgColor()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>556</x>
     <y>103</y>
    </hint>
    <hint type="destinationlabel">
     <x>585</x>
     <y>-10</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>maxSteps</sender>
   <signal>valueChanged(int)</signal>
   <receiver>lcdNumber</receiver>
   <slot>display(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>556</x>
     <y>103</y>
    </hint>
    <hint type="destinationlabel">
     <x>556</x>
     <y>103</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>teCode</sender>
   <signal>textChanged()</signal>
   <receiver>MainWindowClass</receiver>
   <slot>startTimer()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>141</x>
     <y>86</y>
    </hint>
    <hint type="destinationlabel">
     <x>202</x>
     <y>-19</y>
    </hint>
   </hints>
  </connection>
 </connections>
 <slots>
  <slot>compileAndRun()</slot>
  <slot>openCode()</slot>
  <slot>saveCode()</slot>
  <slot>saveImage()</slot>
  <slot>setBkgColor()</slot>
  <slot>startTimer()</slot>
 </slots>
</ui>
